src/ThirdPartyApi/Program.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Hsm.ClientService.ThirdPartyApi.Client.Configuration;
+using Hsm.ClientService.ThirdPartyApi.Client.Extensions;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Polly;
+using Polly.Contrib.WaitAndRetry;
+using Serilog;
+using ThirdPartyApi;
+
+var hostBuilder = Host.CreateDefaultBuilder();
+
+hostBuilder.UseSerilog((context, loggerConfiguration) =>
+    loggerConfiguration.ReadFrom.Configuration(context.Configuration));
+
+hostBuilder.ConfigureServices((context, services) =>
+{
+    var thirdPartyApiConfiguration = context.Configuration
+        .GetSection(nameof(ThirdPartyApiConfiguration))
+        .Get<ThirdPartyApiConfiguration>();
+
+    services.AddThirdPartyApiClient(config =>
+        {
+            config.BaseUrl = thirdPartyApiConfiguration.BaseUrl;
+            config.PersonalAccessToken = thirdPartyApiConfiguration.PersonalAccessToken;
+        },
+        clientBuilder => clientBuilder.AddTransientHttpErrorPolicy(policy =>
+            policy.WaitAndRetryAsync(Backoff.DecorrelatedJitterBackoffV2(TimeSpan.FromSeconds(1), retryCount: 5))));
```

Comment:
I'm not sure if this can be used with the client service as far as I remember they're using 503 sometimes for error codes when other services are failing and as far as I remember this Jitter will retry 503 as well

Reply:
yes it seems so:
![image](https://github.com/Hagleitner-IoT/Hsm.ClientService.Samples/assets/158452641/99278d8c-aa14-475f-a10d-88874d673d39)
i now added a simple check that we only retry if its not "ServiceUnavailable".
should we wrap this into its own extension or something or is it fine here, written explicitly?

Reply:
Fine that way